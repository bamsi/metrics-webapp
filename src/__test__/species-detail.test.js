import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import store from "../redux/configureStore";
import SpeciesDetail from "../components/fish/species-detail";
import { shallow } from "enzyme";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000/species",
    state: { species: "White Hook" },
  }),
}));

describe("Testing rendering of the mission page", () => {
  it("", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          shallow(
          <SpeciesDetail />)
        </Router>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
