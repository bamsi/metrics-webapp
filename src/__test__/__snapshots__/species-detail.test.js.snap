// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing rendering of the mission page  1`] = `
<div>
  shallow(
  <div
    class="loading"
  >
     Loading....
  </div>
  )
</div>
`;
